core/src/main/java/org/mule/api/config/ConfigurationInstanceNotification.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.config;
+
+import org.mule.context.notification.CustomNotification;
+import org.mule.extension.api.runtime.ConfigurationInstance;
+
+/**
+ * Custom notification that communicates the change of state in a {@link ConfigurationInstance}
+ *
+ * @since 4.0
+ */
+public class ConfigurationInstanceNotification extends CustomNotification
+{
+
+    private static final int CONFIGURATION_INSTANCE_ACTION_BASE = (CUSTOM_EVENT_ACTION_START_RANGE + 4) * 5;
+    private static int ACTION_INDEX = 0;
+
+    public static final int CONFIGURATION_BEING_STOPPED = ++ACTION_INDEX + CONFIGURATION_INSTANCE_ACTION_BASE;
```

Comment:
should be CONFIGURATION_STOPPED and fired after the stop() has ended (even if it threw exception)